### PR TITLE
Add service to check material status for impacts and interventions

### PR DESCRIPTION
**Pull request by yulia-bel**
### General description

1. Adding validation service to material service
2. Using service for impact and new intervention endpoints
3. Tests

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Provide minimal instructions on how to test this PR._

- Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging

- [ ] Branch name / PR includes the related Jira ticket Id.
- [ ] Tests to check core implementation / bug fix added.
- [ ] All checks in Continuous Integration workflow pass.
- [ ] Feature functionally tested by reviewer(s).
- [ ] Code reviewed by reviewer(s).
- [ ] Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: api/src/modules/materials/material.repository.ts [hunk 1]
```diff
@@ -112,4 +113,26 @@ export class MaterialRepository extends ExtendedTreeRepository<
...
+      const inactiveMaterialNames: string[] = inactiveSelectedMaterials.map(
+        (material: Material) => material.name,
+      );
+      throw new BadRequestException(
+        `Following Requested Materials are not activated: ${inactiveMaterialNames.join(
+          ', ',
+        )}`,
+      );
+    }
+  }
```
Comment (KevSanchez) [inline api/src/modules/materials/material.repository.ts:137]: See same comment on PR 849
https://github.com/Vizzuality/landgriffon/pull/849#discussion_r1100043333

**Review by KevSanchez — CHANGES_REQUESTED** [review]

File: api/src/modules/materials/material.repository.ts [hunk 2]
```diff
@@ -112,4 +113,26 @@ export class MaterialRepository extends ExtendedTreeRepository<
     // Recursively find elements and their ancestry given Ids of the subquery above
     return this.getEntityAncestry<Material>(queryBuilder, Material.name);
   }
+
+  async checkActiveMaterials(materialIds: string[]): Promise<void> {
+    const inactiveSelectedMaterials: Material[] = await this.dataSource
```
Comment (alexeh) [inline api/src/modules/materials/material.repository.ts:118]: I think this is already being handled, but note that you are using dataSource's `.getRepository()` method to get the material's repository, when you already are in the MaterialRepository class

File: api/src/modules/materials/material.repository.ts (hunk 1 above)
Comment (yulia-bel) [reply api/src/modules/materials/material.repository.ts]: fixed, same as in indicators :+1:

File: api/src/modules/materials/material.repository.ts (hunk 2 above)
Comment (yulia-bel) [reply api/src/modules/materials/material.repository.ts]: yep, thanks, all logic moved to service

**Review by alexeh — APPROVED** [review]
Niceee, thanks @yulia-bel !

**Review by KevSanchez — APPROVED** [review]

**Pull request merged**